### ISSUE TITLE
preact-resnet50: Got loss NaN suddenly.

### ISSUE THREAD

**Issue by zeakey** [NVIDIA/DALI#1377]
I'm training the preacttion resnet50 (also know as resnet50_v2 in gluoncv) with the official example training script ([docs/examples/pytorch/resnet50/main.py](https://github.com/NVIDIA/DALI/blob/master/docs/examples/pytorch/resnet50/main.py)).

The command I used to train is `python -m torch.distributed.launch --nproc_per_node=4 main.py /media/ssd0/ilsvrc12/ --fp16 --b 64 --static-loss-scale 128.0 --workers 2  2>&1 | tee log.txt`

The network definition is here <https://paste.ubuntu.com/p/RDKcqcdf4V/> which follows a `BN-ReLU-Conv` pipeline.
The only difference between my implementation and [gluoncv.model_zoo.resnet50_v2](https://github.com/dmlc/gluon-cv/blob/f6a97bacb3ed0174c66117f329f2961eea610f5a/gluoncv/model_zoo/resnet.py#L427-L502) is that gluoncv implementation has a batchnorm layer at the begining of the network, which I think is neglectable because the input data has already been whitten.

Everything is fine at the begining. However, the loss strangely turns to NaN after 15-20 epochs iteration. 

Here is my [log](http://data.kaizhao.net/logs/dali-example-preact-resnet50-loss-nan.log).

I've trained with `--fp16` twice and both of them crashed with 25 epochs. Now I'm training without `--fp16` and currently it has run 45 epochs, everything seems good.

Why did the loss turn to NaN at the middle of training? Should I tune the `--static-loss-scale 128.0` to a smaller value?

**Comment by JanuszL** [comment]
Hi,
Have you tried to train your network without DALI?

The RN50 is an example showing how to use DALI, but it is not meant to experiment with different training options. For that, I would strongly recommend going to our official Nvidia examples - https://github.com/NVIDIA/DeepLearningExamples.

If it still doesn't work feel free to ask a question [there](https://github.com/NVIDIA/DeepLearningExamples/issues).

**Comment by zeakey** [comment]
@JanuszL Hi, as descripted in the issue, I turned off '--fp16' and everything workds smoothly.
I guest if the `--static-loss-scale 128.0` argument is too big that the gradients overflowed during backward?

Should I try to decrease `--static-loss-scale 128.0` (e.g. to 64)?

**Comment by JanuszL** [comment]
In our CI we are using:
```
python -m torch.distributed.launch --nproc_per_node=${NUM_GPUS} main.py -a resnet50 --dali_cpu --fp16 --b 128 --static-loss-scale 128.0 --workers 4 --lr=0.4
```
Where `NUM_GPUS=8` and it fully trains. As said, please check with https://github.com/NVIDIA/DeepLearningExamples as DALI examples were not tested with different hyperparameters.

**Comment by JanuszL** [comment]
If it still doesn't work please reopen.
